runtime/ts/relevance.ts

```diff
@@ -41,17 +44,19 @@ export class Relevance {
 
   // Returns false, if at least one of the particles relevance lists ends with a negative score.
   isRelevant(plan) {
-    let hasUi = plan.particles.some(p => Object.keys(p.consumedSlotConnections).length > 0);
+    const hasUi = plan.particles.some(p => Object.keys(p.consumedSlotConnections).length > 0);
     let rendersUi = false;
     this.relevanceMap.forEach((rList, particle) => {
       if (rList[rList.length - 1] < 0) {
         return false;
       } else if (Object.keys(particle.consumedSlotConnections).length) {
         rendersUi = true;
+        return false;
       }
+      return false;
```

Comment:
The return value has no effect in a forEach callback, and since the loop only ever latches rendersUi true, we should change it to a regular loop and break as soon as `(rList[rList.length - 1] >= 0 && Object.keys(particle.consumedSlotConnections).length)`.